src/utils/ValidatorUtils.ts

```diff
@@ -0,0 +1,29 @@
+import { BigNumber, ethers } from "ethers";
+import { object, number as Number, min as Min, define, optional, string } from "superstruct";
+import { Deposit } from "../interfaces";
+
+const AddressValidator = define<string>("AddressValidator", (v) => ethers.utils.isAddress(String(v)));
+const BigNumberValidator = define<BigNumber>("BigNumberValidator", (v) => ethers.BigNumber.isBigNumber(v));
+
+const DepositSchema = object({
+  depositId: Min(Number(), 0),
+  depositor: AddressValidator,
+  recipient: AddressValidator,
+  originToken: AddressValidator,
+  amount: BigNumberValidator,
+  originChainId: Min(Number(), 0),
```

Comment:
Can we make all these `Number` --> `Integer`?

Reply:
Done: https://github.com/across-protocol/sdk-v2/pull/428/commits/fe9d4251b9ca527bcbf384f1525ed895a8d72730